lib/graphql/query.rb

```diff
@@ -38,11 +30,19 @@ def call(member)
     # @param max_depth [Numeric] the maximum number of nested selections allowed for this query (falls back to schema-level value)
     # @param max_complexity [Numeric] the maximum field complexity for this query (falls back to schema-level value)
     # @param except [<#call(schema_member)>] If provided, objects will be hidden from the schema when `.call(schema_member)` returns truthy
-    def initialize(schema, query_string = nil, document: nil, context: nil, variables: {}, validate: true, operation_name: nil, root_value: nil, max_depth: nil, max_complexity: nil, except: NullExcept)
+    def initialize(schema, query_string = nil, document: nil, context: nil, variables: {}, validate: true, operation_name: nil, root_value: nil, max_depth: nil, max_complexity: nil, except: nil, only: nil)
       fail ArgumentError, "a query string or document is required" unless query_string || document
 
       @schema = schema
-      @warden = GraphQL::Schema::Warden.new(schema, except)
+      mask = if except
+        wrap_if_legacy_mask(except)
+      elsif only
+        InvertedMask.new(wrap_if_legacy_mask(only))
```

Comment:
Thank you for doing this. 👍 We ended up inverting the `except` logic in GitHub's implementation since it was easier to reason with and fit the model we wanted better.